### DISCUSSION TITLE
第4节课作业(13班)

### DISCUSSION THREAD

**Discussion by Lum1104** [InternLM/Tutorial#259] [作业]
1. 基础作业：

构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手，效果如下图所示，本作业训练出来的模型的输出需要将不要葱姜蒜大佬替换成自己名字或昵称！

2. 进阶作业：

将训练好的Adapter模型权重上传到 OpenXLab、Hugging Face 或者 MoelScope 任一一平台。
将训练好后的模型应用部署到 OpenXLab 平台，参考部署文档请访问：https://aicarrier.feishu.cn/docx/MQH6dygcKolG37x0ekcc4oZhnCe

**整体实训营项目：**

时间周期：即日起致课程结束

即日开始可以在班级群中随机组队完成一个大作业项目，一些可提供的选题如下：

人情世故大模型：一个帮助用户撰写新年祝福文案的人情事故大模型
中小学数学大模型：一个拥有一定数学解题能力的大模型
心理大模型：一个治愈的心理大模型
工具调用类项目：结合 Lagent 构建数据集训练 InternLM 模型，支持对 MMYOLO 等工具的调用
其他基于书生·浦语工具链的小项目都在范围内，欢迎大家充分发挥想象力。

作业[参考答案](https://github.com/InternLM/tutorial/blob/main/xtuner/self.md)

**Comment by SiliconSisyphus** [comment]
<img width="1140" alt="image" src="https://github.com/InternLM/tutorial/assets/152180631/ffbcb8b8-2c45-4b2c-ab62-967a4b7f1f68">
好像在web_demo.py运行的时候访问出错了，查看了文件夹，还在找问题

**Comment by qingyin741** [comment]
![Snipaste_2024-01-15_16-30-48](https://github.com/InternLM/tutorial/assets/118804949/67a32acb-5e69-4c76-8fab-4bae4f1661c3)

**Comment by BOOMO4** [comment]
https://l055jh6h54r.feishu.cn/wiki/MmHWwBChUizAABk8NJCcw5t5nTf?from=from_copylink

**Comment by Xrcb** [comment]
https://blog.csdn.net/weixin_64546620/article/details/135597127
